programming/editors/vscode.md

```diff
@@ -0,0 +1,203 @@
+---
+layout: page
+title: VSCode
+---
+
+# Visual Studio Code
+
+Visual Studio Code (known commonly as VSCode) is a free code editor developed by
+Microsoft. As such it has [extensive documentation][vscode-docs] about its
+general use, which we will not repeat here. Instead this page will focus on the
+ways to get the most out of VSCode when developing code for Student Robotics.
+
+[vscode-docs]: https://code.visualstudio.com/docs
+
+## Python Extension
+
+While VSCode has some built-in support for developing Python, we recommend that
+you additionally install the official [Python extension][ms-python.python]. This
+brings more comprehensive support and enables everything else we will document
+here.
+
+[ms-python.python]: https://marketplace.visualstudio.com/items?itemName=ms-python.python
+
+## Code Completions
+
+In order for VSCode to pick up the `sr.robot` library and offer completions
```

Comment:
A fresh install of VSCode prompted me to install pylint, if I hit yes the instructions described here still result in a pylint linting error. Should we recommend not to install pylint or include instructions to fix pylint?

Reply:
I agree it's not optimal, however I'm most tempted to ignore this. My reasoning (as discussed in person) is that the popup can also appear whenever the user opens a Python file and is sufficiently out of the way of the flow we're taking them through here that I don't think it's likely to block them from using this. We definitely could walk people through configuring pylint, though I suspect that would fit better in a separate section on linters rather than as part of the editor configs.